### ISSUE TITLE
Artemis displays incorrect bases in DNA view

### ISSUE THREAD

**Issue by anishazaveri** [sanger-pathogens/Artemis#310]
I'm using Artemis 18.0.1 on Windows 10 with Java version "1.8.0_241". 
I see this error across two machines with the same specifications and with multiple .gbk/.fasta files

I search for a given sequence, and it claims to have found it, but the highlighted string does not match the search query

![image](https://user-images.githubusercontent.com/42725952/82471102-3e57cc00-9a94-11ea-913f-fa3adb5b6cb4.png)

Next, I attempted to 'View the bases of selection as Fasta'

![image](https://user-images.githubusercontent.com/42725952/82471261-78c16900-9a94-11ea-91df-81bc8e86d9cd.png)

This shows the same result as the query. The issue appears to be in the display of the DNA Viewer!

A similar problem: I use the navigator to go to base 5606. It's unclear from the DNA Viewer what this base is since the highlighted area doesn't clearly highlight a single base, but it appears to be a 'T'

![image](https://user-images.githubusercontent.com/42725952/82471722-1452d980-9a95-11ea-8826-f211a02e475a.png)

Now, I select that base as fasta:
And it says it's a G
![image](https://user-images.githubusercontent.com/42725952/82471775-2cc2f400-9a95-11ea-8a05-865d68486b11.png)

**Comment by MatthiasHBeck** [comment]
I'm having the exact same issue in both Artemis 18.0.2 and 18.1.0
and I'm using a fresh installation of 

`openjdk version "11.0.8" 2020-07-14`
`OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.8+10)`
`OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.8+10, mixed mode)`

Because I thought previously installed JDK jdk-14.0.2
was causing that issue.
But that probably was not the case since the problem still occurs after uninstalling JDK 14 and reinstalling the recommended JDK 11.
Any help is very much appreciated.

**Comment by ireneortega** [comment]
Same as @anishazaveri and @MatthiasHBeck happens for me. 

Even more, bases position bar has moved -3 pb so every gen starts three nucleotide before the real starting site (except for the first gen). Although the fasta sequences are right, the sequences highlighted are not. Here is an example of what happens. The protein sequence starts with M, but Artemis highlightes the previous aminoacid T.

![image](https://user-images.githubusercontent.com/58036036/98472139-4f86c780-21f1-11eb-9000-a153cb4d2d77.png)

How can I solve it?

**Comment by PrebenKrabben** [comment]
Very frustrating error. I have the same issue.

I first thought the issue was related to Artemis version 18 as I recently up-graded. I had to install AdoptOpenJDK JRE 11.0.9.101 in order to run Artemis version 18. However running Artemis version 16 under AdoptOpenJDK JRE 11.0.9.101 shows that the issue was already there in version 16. Next idea was that the issue is with AdoptOpenJDK JRE 11.0.9.101 and I un-installed it and re-installed "Java(TM) SE Runtime Environment (build 1.8.0_271-b09)". However, the issue still remains in version 16.

What is even worse is that if I want to mark up a new DNA-feature you cannot directly select the bases, but I need to go in and View/Bases to see which bases I have selected.

I don't know if this could be an Windows issue or not, but I'm running Windows 10

**Comment by kw2990** [comment]
I also have the same issue. Please help, it's very frustrating!

**Comment by PrebenKrabben** [comment]
> I also have the same issue. Please help, it's very frustrating!

I resorted to have both Artemis version 18 and version 6 installed. Version 6 is the last version without this problem.

**Comment by kw2990** [comment]
> > I also have the same issue. Please help, it's very frustrating!
> 
> I resorted to have both Artemis version 18 and version 6 installed. Version 6 is the last version without this problem.

I have managed a work around - I used this jarfix.exe (https://johann.loefflmann.net/en/software/jarfix/index.html) with the additional config file for JDK. This has allowed me to use Artemis v15. Weirdly the text is zoomed out when you open Artemis, but is still readable.

**Comment by MatthiasHBeck** [comment]
> 
> 
> > > I also have the same issue. Please help, it's very frustrating!
> > 
> > 
> > I resorted to have both Artemis version 18 and version 6 installed. Version 6 is the last version without this problem.
> 
> I have managed a work around - I used this jarfix.exe (https://johann.loefflmann.net/en/software/jarfix/index.html) with the additional config file for JDK. This has allowed me to use Artemis v15. Weirdly the text is zoomed out when you open Artemis, but is still readable.

Many thanks kw2990 for posting a solution! I've tried this, unfortunately without success. Maybe I used the wrong configuration in the config file? 
I used 

` action="C:\Java\JDKs\JDK12-early-build\bin\javaw.exe" -jar "%1" %*` 

with the path of my current JDK javaw.exe file...

But the error still remains. 
Can you tell me what configurations you used exactly? 
Thank you. 

P.S.: Another not very delicate work around: I noticed when you have a second monitor connected and use this one as a second screen the bases are displayed correctly again in Artemis.

**Comment by kw2990** [comment]
> > > > I also have the same issue. Please help, it's very frustrating!
> > > 
> > > 
> > > I resorted to have both Artemis version 18 and version 6 installed. Version 6 is the last version without this problem.
> > 
> > 
> > I have managed a work around - I used this jarfix.exe (https://johann.loefflmann.net/en/software/jarfix/index.html) with the additional config file for JDK. This has allowed me to use Artemis v15. Weirdly the text is zoomed out when you open Artemis, but is still readable.
> 
> Many thanks kw2990 for posting a solution! I've tried this, unfortunately without success. Maybe I used the wrong configuration in the config file?
> I used
> 
> ` action="C:\Java\JDKs\JDK12-early-build\bin\javaw.exe" -jar "%1" %*`
> 
> with the path of my current JDK javaw.exe file...
> 
> But the error still remains.
> Can you tell me what configurations you used exactly?
> Thank you.
> 
> P.S.: Another not very delicate work around: I noticed when you have a second monitor connected and use this one as a second screen the bases are displayed correctly again in Artemis.

Sorry MatthiasHBeck, I honestly can't remember how I got it to work. As far as I can tell the config file that I used is unchanged from the one on the jarfix website - I think maybe I just downloaded it to the same location as jarfix.exe and then from the command line ran "jarfix /c"?

**Comment by MatthiasHBeck** [comment]
> Sorry MatthiasHBeck, I honestly can't remember how I got it to work. As far as I can tell the config file that I used is unchanged from the one on the jarfix website - I think maybe I just downloaded it to the same location as jarfix.exe and then from the command line ran "jarfix /c"?

Thanks a million for getting back to us here. I’ll try again and post in case I could make it work as well.

**Comment by Phytobacteriology-UPNA** [comment]
This same thing happened to me. I linked it to having the size of text (in Display settings) changed to 125% or higher. If you put the scale back to 100%, it should work. It will also work if you use Artemis v. 17 or lower with Java 8 or lower. I do not have a good solution for this, except changing the size of text in the display to 100%, but then the text is way to small for my eyes.

Is this also the cause of your problem?

Cheers,
Jesús

![Screen Capture](https://user-images.githubusercontent.com/59928545/111202790-93854100-85c4-11eb-9abd-4452faf8f73a.PNG)

**Comment by Phytobacteriology-UPNA** [comment]
I forgot to say that I also have Windows 10, and that I resorted to remain working with Artemis v 17 and Java 8

**Comment by Phytobacteriology-UPNA** [comment]
> 
> 
> I'm using Artemis 18.0.1 on Windows 10 with Java version "1.8.0_241".
> I see this error across two machines with the same specifications and with multiple .gbk/.fasta files
> 
> I search for a given sequence, and it claims to have found it, but the highlighted string does not match the search query
> 
> ![image](https://user-images.githubusercontent.com/42725952/82471102-3e57cc00-9a94-11ea-913f-fa3adb5b6cb4.png)
> .......


I got a helpful solution from William Roberts-Sengier, from the Artemis Helpdesk. I quote:

"There is an option to change the font size in Artemis using an options file (as described on https://sanger-pathogens.github.io/Artemis/Artemis/artemis-manual.html#_The_Options_File).

To do this, you simply need to create a file called "options" or "options.txt" in the same directory as artemis.jar. In that file, add a line: font_size = 20

You can adjust the font size to whatever suits your needs and the Window's display settings. I found that when I set Windows display to 125% (via Settings -> Ease of Access -> Display -> Make everything bigger -> 125%), font size 20 worked well in Artemis. I have attached a screenshot showing the difference between font size 20 (left) and size 12 (right), with the Windows display setting at 125%.

Increasing the font size too much can make it harder to read text in certain parts of the program, e.g. the File Manager, so I can only advise you to experiment until you find a setting that works best for you."

For me, using Artemis 18-Java 11-Display text 125% in a 27" monitor, I found it fixed with font_size=13
It might be different in your system, though. An unwanted consequence is that all the panes in the Artemis main window also get larger, so the bottom panel with the list of features gets very small can can even dissappear.

In summary, there are four ways to fix this problem:

- Set the display text size to 100% (although the resulting small text of Windows can be hard on your eyes)
- Lower the resolution of the display. The text will be larger, but it might get blurry. You'll also see an enlarged Artemis window, which fits less items
- Use Artemis v 17 with Java v. 8
- Use the solution provided by William, which allows you to uses the latest version of Artemis and Java

Good luck. Cheers,
Jesús

**Comment by Tommy-TOMTOM** [comment]
Hi !  Have you solved that issue ? I have the same problem with displaying wrong bases...

**Comment by Phytobacteriology-UPNA** [comment]
> Hi ! Have you solved that issue ? I have the same problem with displaying wrong bases...

Hi TOMTOM, the four solutions that worked for me are those in the above post (March 18, 2021). You might have to play adjusting the font size in the options file, but using font 13 worked fine by me. I created an options file (attached here) adjusting the font size to 13, and put the file in the same folder where Artemis is initiated. It worked well in different monitors; hope it does for you too.
Cheers, J

[options.txt](https://github.com/sanger-pathogens/Artemis/files/14883890/options.txt)

**Comment by Tommy-TOMTOM** [comment]
> Hi TOMTOM, the four solutions that worked for me are those in the above post (March 18, 2021). You might have to play adjusting the font size in the options file, but using font 13 worked fine by me. I created an options file (attached here) adjusting the font size to 13, and put the file in the same folder where Artemis is initiated. It worked well in different monitors; hope it does for you too. Cheers, J
> 
> [options.txt](https://github.com/sanger-pathogens/Artemis/files/14883890/options.txt)

Hi Phytobacteriology-UPNA, that's amazing, font size solved my problem. Thank you for prepared options file, it really works !
